src/content/popup/KanjiReferencesTable.tsx

```diff
@@ -0,0 +1,102 @@
+import { KanjiResult } from "@birchill/jpdict-idb";
+
+import { PopupOptions } from "./popup";
+import { getSelectedReferenceLabels } from "../../common/refs";
+import { getReferenceValue } from "../reference-value";
+import { I18nProvider, useLocale } from "../../common/i18n";
+
+type Props = {
+  entry: KanjiResult,
+  options: PopupOptions,
+}
+
+// temporary hack
+export function WrappedKanjiReferencesTable(props: Props) {
+  return (
+    <I18nProvider>
+      <KanjiReferencesTable {...props} />
+    </I18nProvider>
+  )
+}
+
+export function KanjiReferencesTable({ entry, options }: Props) {
+  const { t } = useLocale();
+  const referenceNames = getSelectedReferenceLabels(options.kanjiReferences, t);
```

Comment:
I think we could grab all this logic and stick it in a `useMemo` hook keyed on `options.kanjiReferences` and `t` etc.

Reply:
Thanks! I think we could actually take the whole block of code that calculates `referenceTableInfo` and `numReferences` and wrap that up in the `useMemo` hook. Then again, we currently never re-render this component so using `useMemo` at all is probably premature optimization.